### PR TITLE
Add support for parsing torch.Generator in JIT

### LINKED ISSUE #140420: TorchScript IR Parser does not support `torch.Generator`
### 🐛 Describe the bug

Support for `torch.Generator` in TorchScript was added in https://github.com/pytorch/pytorch/pull/110413. However, something that was missed was support for parsing `torch.Generator` from a serialized string representation of the JIT graph, e.g.
```python
import torch

torch._C.parse_ir(
    """
graph():
  %0 : float = prim::Constant[value=-0.31622776601683789]()
  %1 : float = prim::Constant[value=0.31622776601683789]()
  %2 : Generator = prim::Constant[value=torch.Generator(device="cpu", seed=352461024221769975)]()
  %3 : NoneType = prim::Constant()
  %4 : int[] = prim::Constant[value=[10, 10]]()
  %5 : int = prim::Constant[value=6]()
  %6 : Device = prim::Constant[value="cpu"]()
  %7 : Tensor = aten::empty(%4, %5, %3, %6, %3, %3)
  %8 : Float(10, 10) = aten::uniform(%7, %0, %1, %2)
  return (%8)
    """,
)
```
results in the following runtime error
```
RuntimeError: expected ] but found '.' here:
  %0 : float = prim::Constant[value=-0.31622776601683789]()
  %1 : float = prim::Constant[value=0.31622776601683789]()
  %2 : Generator = prim::Constant[value=torch.Generator(device="cpu", seed=352461024221769975)]()
                                             ~ <--- HERE
  %3 : NoneType = prim::Constant()
  %4 : int[] = prim::Constant[value=[10, 10]]()
```

### Versions

```
PyTorch version: 2.4.0+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: CentOS Linux release 7.9.2009 (Core) (x86_64)
GCC version: Could not collect
Clang version: 3.4.2 (tags/RELEASE_34/dot2-final)
CMake version: version 3.27.1
Libc version: glibc-2.30

Python version: 3.8.16 (default, Mar 18 2024, 18:27:40)  [GCC 8.4.0] (64-bit runtime)
Python platform: Linux-3.10.0-1160.80.1.el7.x86_64-x86_64-with-glibc2.2.5
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                4
On-line CPU(s) list:   0-3
Thread(s) per core:    2
Core(s) per socket:    2
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 85
Model name:            Intel(R) Xeon(R) Platinum 8259CL CPU @ 2.50GHz
Stepping:              7
CPU MHz:               2499.998
BogoMIPS:              4999.99
Hypervisor vendor:     KVM
Virtualization type:   full
L1d cache:             32K
L1i cache:             32K
L2 cache:              1024K
L3 cache:              36608K
NUMA node0 CPU(s):     0-3
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single rsb_ctxsw fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid mpx avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 ida arat pku ospke

Versions of relevant libraries:
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.24.4
[pip3] torch==2.4.0+cpu
[pip3] torchvision==0.19.0+cpu
[conda] No relevant packages
```

cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

### PR DESCRIPTION
**Pull request by antoniojkim**
Fixes #140420

cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -14148,6 +14148,34 @@ def test(tensor, generator):
 
             FileCheck().check_not("prim::PythonOp").run(cu.test.graph)
 
+    def test_parse_generator(self):
+        seed = 2024
```
Comment (davidberard98) [inline test/test_jit.py:14152]: iiuc seed can be in [-2^63, 2^64) or perhaps [0, 2^64). Can you test some large seeds to see if the parser works?

e.g.:
* if you serialize a torchscript graph for a generator with a negative seed, is its value remapped to a positive value?
* can the parser work with a seed > 2^64 ?

**Review by davidberard98 — COMMENTED** [review]
LGTM, can you add a few more tests though?

File: test/test_jit.py (hunk 1 above)
Comment (antoniojkim) [reply test/test_jit.py:14152]: Technically seed can be `[0, 2^64)`. However, the scalar parser only supports signed integers. So, the parser is constrained to `[0, 2^63)`. However, that is good enough for my purposes. I've added tests for these cases

**Review by davidberard98 — APPROVED** [review]

**Pull request closed without merging**